src/io/leopard/toLeopard.ts

```diff
@@ -253,7 +422,41 @@ export default function toLeopard(
   function blockToJSWithContext(block: Block, target: Target, script?: Script): string {
     return blockToJS(block);
 
-    function inputToJS(input: BlockInput.Any): string {
+    function increase(leftSide: string, input: BlockInput.Any, allowIncrementDecrement: Boolean) {
+      let n;
+      if (input.type === "block" || ((n = Number(input.value)), isNaN(n))) {
+        return `${leftSide} += (${inputToJS(input, "number")});`;
+      }
+
+      if (allowIncrementDecrement && n === 1) {
+        return `${leftSide}++;`;
+      } else if (allowIncrementDecrement && n === -1) {
+        return `${leftSide}--;`;
+      } else if (n >= 0) {
+        return `${leftSide} += ${JSON.stringify(n)};`;
+      } else if (n < 0) {
+        return `${leftSide} -= ${JSON.stringify(-n)};`;
+      }
+    }
+
+    function decrease(leftSide: string, input: BlockInput.Any, allowIncrementDecrement: Boolean = true) {
+      let n;
+      if (input.type === "block" || ((n = Number(input.value)), isNaN(n))) {
```

Comment:
How many places are we doing an "input is a primitive, and its value is X type" check? It might be worth abstracting it into a function.